### PR TITLE
backend/oss: Support for assume role config

### PR DESCRIPTION
**Pull request by xiaozhu36**
This PR aims to support assume_role config in the Alibaba Cloud backend. Currently, the terraform-provider-alicloud has supported it and the backend also need to do it.

In addition, this PR also aims to fix the issue: [terraform-provider-alicloud/issues#1385](https://github.com/terraform-providers/terraform-provider-alicloud/issues/1385)

### CODE REVIEW

File: backend/remote-state/oss/backend.go [hunk 1]
```diff
@@ -137,6 +142,42 @@ func New() backend.Backend {
...
+				"session_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "The session name to use when assuming the role.",
+					DefaultFunc: schema.EnvDefaultFunc("ALICLOUD_ASSUME_ROLE_SESSION_NAME", "terraform"),
+				},
+				"policy": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "The permissions applied when assuming a role. You cannot use, this policy to grant further permissions that are in excess to those of the, role that is being assumed.",
```
Comment (teamterraform) [inline backend/remote-state/oss/backend.go]: A small nitpick - the commas are unnecessary:
```suggestion
					Description: "The permissions applied when assuming a role. You cannot use this policy to grant permissions which exceed those of the role that is being assumed.",
```

**Comment by teamterraform** [comment]
Thank you for working on this, @xiaozhu36 ! I had a small grammatical nitpick, but once that's fixed I will approve and merge. Cheers!

**Comment by xiaozhu36** [comment]
HI @teamterraform Thanks for your feedback. I have fixed it and please check it again.

**Comment by teamterraform** [comment]
Thank you very much!

**Pull request merged**